### DISCUSSION TITLE
How can a service call another service

### DISCUSSION THREAD

**Discussion by Akronae** [krojew/springtime#2] [Q&A] [answered]
Lets say I have a Booking service, how can that Booking service call let's say User service?  
In other word how can I DI into some services?

**Comment by krojew** [comment, marked as answer]
You can do this in two ways, depending whether you want to inject a trait or a concrete type. With a concrete type, you'll have something like this:

```
#[derive(Component)]
struct UserService {
}

#[derive(Component)]
struct BookingService {
    user_service: ComponentInstancePtr<UserService>,
}
```

With traits:

```
#[injectable]
trait UserService {
}

#[derive(Component)]
struct ConcreteUserService {
}

#[component_alias]
impl UserService for ConcreteUserService {
}

#[derive(Component)]
struct BookingService {
    user_service: ComponentInstancePtr<dyn UserService + Send + Sync>,
}
```

For more information, you can take a look at the tutorial/examples: https://github.com/krojew/springtime/tree/master/springtime-di/examples

**Reply by Akronae** [reply]
This is great thanks! I looked at the examples too quickly
